src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -0,0 +1,368 @@
+     * @return The map name as a c style char string.
+     *
+     * @note This is only available through the LocationStructure object (no access from base structure).
+     */
+    const char * name_c_str() const
+    {
+        // note: if name is null, this will return an empty terminated c string - ""
+        return mLocationNameBuffer;
+    }
+
+private:
+    // extra char to allow for terminating 0 ( or name_c_str())
+    char mLocationNameBuffer[kLocationNameMaxSize + 1];
+};
+
+/**
+ * This class wraps the MapStruct object and provides a more user-friendly interface for the data.
+ */
+struct MapStructureWrapper : public chip::app::Clusters::ServiceArea::Structs::MapStruct::Type
+{
+    /**
+     * @brief This is a default constructor that initializes the map object with default values.
+     */
+    MapStructureWrapper() { Set(0, CharSpan()); }
+
+    /**
+     * @brief This is a full constructor that initializes the map object with the given values. All values are deep copied.
+     * @param[in] aMapId The identifier of this map.
+     * @param[in] aMapName A human readable name (should not be empty string).
+     *
+     * @note Requirements regarding what combinations of fields and values are 'valid' are not checked by this class.
+     * @note If aMapName is larger than kMapNameMaxSize, it will be truncated.
+     */
+    MapStructureWrapper(uint8_t aMapId, const CharSpan & aMapName) { Set(aMapId, aMapName); }
+
+    /**
+     * @brief This is a copy constructor that initializes the map object with the values from another map object. All values are
+     * deep copied.
+     * @param[in] aOther The map object to copy.
+     */
+    MapStructureWrapper(const MapStructureWrapper & aOther) { *this = aOther; }
+
+    /**
+     * @brief This is an assignment operator that initializes the map object with the values from another map object. All values are
+     * deep copied.
+     * @param[in] aOther The map object to copy.
+     */
+    MapStructureWrapper & operator=(const MapStructureWrapper & aOther)
+    {
+        Set(aOther.mapID, aOther.name);
+        return *this;
+    }
+
+    /**
+     * @brief Set all fields of the mop object. All values are deep copied.
+     * @param[in] aMapId The identifier of this map.
+     * @param[in] aMapName A human readable name (should not be empty string).
+     *
+     * @note Requirements regarding what combinations of fields and values are 'valid' are not checked by this class.
+     * @note if aMapName is larger than kMapNameMaxSize, it will be truncated.
+     */
+    void Set(uint8_t aMapId, const CharSpan & aMapName)
+    {
+        mapID = aMapId;
+
+        // Save the map name to the buffer so that access and lifetime is controlled.
+        // Create a CharSpan reference the text.
+        // Make sure there is always a terminating character, mainly so we can get c_str for logging.
+        memset(mMapNameBuffer, 0, sizeof(mMapNameBuffer + 1));
```

Comment:
same bug here: sizeof is wrong. You need to use different functions to handle C-style strings guaranteeing null termination.

Please move these implementations in a cpp file instead of keeping everything in a header.

Reply:
Implementation in h file follows example set by operational-state-cluster-objects.h

maximum size NON terminated string = kMapNameMaxSize
maximum size termainated string = kMapNameMaxSize + 1 = sizeof(mMapNameBuffer)

Reply:
Fixed [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/e2ef26e7de102eae3bce6b1164384360702ea38e).